python/whylogs/core/schema.py

```diff
@@ -161,6 +173,30 @@ def _resolve_pdf(self, df: pd.DataFrame, force_resolve: bool = False) -> bool:
 
         return dirty
 
+    def _augment(
```

Comment:
Maybe just make *this* the callback...

Reply:
This operation is a pre-profiling transform step, seems like it should live in Preprocessing conceptually. Having the pandas / row split in the signature is a leaky abstraction, it requires that the integrator know about this internal detail of how we split the inputs. Why not just pass the original data before the split, or we could pass them the extracted column where bound?